flink-dist/src/main/flink-bin/bin/pyflink-gateway-server.sh

```diff
@@ -67,4 +67,16 @@ if [[ -n "$FLINK_TESTING" ]]; then
   done < <(find "$FLINK_SOURCE_ROOT_DIR" ! -type d \( -name 'flink-*-tests.jar' -o -path "${FLINK_SOURCE_ROOT_DIR}/flink-connectors/flink-connector-elasticsearch-base/target/flink*.jar" -o -path "${FLINK_SOURCE_ROOT_DIR}/flink-connectors/flink-connector-kafka-base/target/flink*.jar" \) -print0 | sort -z)
 fi
 
-exec $JAVA_RUN $JVM_ARGS "${log_setting[@]}" -cp ${FLINK_CLASSPATH}:${TABLE_JAR_PATH}:${PYTHON_JAR_PATH}:${FLINK_TEST_CLASSPATH} ${DRIVER} ${ARGS[@]}
+COUNT=${#ARGS[@]}
+if [[ ${ARGS[0]} == "local" ]]; then
+  ARGS=("${ARGS[@]:1:$COUNT}")
+  COUNT=$((COUNT-1))
+  if [[ COUNT -gt 1 ]] && [[ ${ARGS[0]} == "-a" ]]; then
+    EXTERAL_CLASSPATH=${ARGS[1]}
+    ARGS=("${ARGS[@]:2:$COUNT}")    
+  fi
+  exec $JAVA_RUN $JVM_ARGS "${log_setting[@]}" -cp ${FLINK_CLASSPATH}:${TABLE_JAR_PATH}:${PYTHON_JAR_PATH}:${FLINK_TEST_CLASSPATH}:${EXTERAL_CLASSPATH} ${DRIVER} ${ARGS[@]}
+else
+  ARGS=("${ARGS[@]:1:$COUNT}")
+  exec "$FLINK_BIN_DIR"/flink run ${ARGS[@]} -c ${DRIVER} -a ${PYTHON_JAR_PATH} -j ${TABLE_JAR_PATH}
```

Comment:
What about removing the option `-a`?
We can set the `isPython` to true if the entrypoint class is "PythonGatewayServer" in `ProgramOptions`.

Reply:
Yes.It sounds better.